<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CooperationPage</class>
 <widget class="QWidget" name="CooperationPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>508</width>
    <height>540</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QLabel" name="headerLabel">
     <property name="text">
      <string>&lt;b&gt;Configure cooperation settings&lt;/b&gt;</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line11">
     <property name="frameShape">
      <enum>QFrame::HLine</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="serverGroup">
     <property name="title">
      <string>Server</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0" colspan="3">
       <widget class="QCheckBox" name="autostartCheckBox">
        <property name="toolTip">
         <string>Select to start the server automatically</string>
        </property>
        <property name="text">
         <string>Start server automatically</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Server Port:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="serverPortSpin">
        <property name="toolTip">
         <string>Enter the port number to listen on</string>
        </property>
        <property name="minimum">
         <number>1025</number>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>42000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>326</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="0" colspan="3">
       <widget class="QCheckBox" name="otherPortsCheckBox">
        <property name="toolTip">
         <string>Select to incrementally try other ports for the server</string>
        </property>
        <property name="text">
         <string>Try other ports for server</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_2">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>No. ports to try:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="portToTrySpin">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Enter the maximum number of additional ports to try</string>
        </property>
        <property name="minimum">
         <number>10</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>326</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="connectionsGroup">
     <property name="title">
      <string>Connections</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="autoAcceptCheckBox">
        <property name="toolTip">
         <string>Select to accept incomming connections automatically</string>
        </property>
        <property name="text">
         <string>Accept connections automatically</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="bannedUsersGroup">
     <property name="title">
      <string>Banned Users</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0" colspan="3">
       <widget class="QListWidget" name="bannedUsersList">
        <property name="selectionMode">
         <enum>QAbstractItemView::ExtendedSelection</enum>
        </property>
        <property name="sortingEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="deleteBannedUsersButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Delete the selected entries from the list of banned users</string>
        </property>
        <property name="text">
         <string>Delete</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="bannedUserEdit">
        <property name="toolTip">
         <string>Enter the user and host of the banned user</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="addBannedUserButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Add the user to the list of banned users</string>
        </property>
        <property name="text">
         <string>Add</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>autostartCheckBox</tabstop>
  <tabstop>serverPortSpin</tabstop>
  <tabstop>otherPortsCheckBox</tabstop>
  <tabstop>portToTrySpin</tabstop>
  <tabstop>autoAcceptCheckBox</tabstop>
  <tabstop>bannedUsersList</tabstop>
  <tabstop>deleteBannedUsersButton</tabstop>
  <tabstop>bannedUserEdit</tabstop>
  <tabstop>addBannedUserButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>otherPortsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_2</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>54</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>51</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>otherPortsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>portToTrySpin</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>161</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
